### DISCUSSION TITLE
SSD1322 u8x8_byte_4wire_hw_spi

### DISCUSSION THREAD

**Discussion by nibble-bit-dev** [olikraus/u8g2#1432] [Q&A]
Hi,
I'm using 'u8x8_byte_4wire_sw_spi' but I saw that the I/O operations are slow.

There is an equivalent; u8x8_byte_4wire_hw_spi  implementation?

Thanks
nbdev

**Comment by olikraus** [comment]
> There is an equivalent; u8x8_byte_4wire_hw_spi implementation?

Yes, but only in the C++ wrapper for Arduino. 
In general this is called "hw_spi" because it depends on your hardware. You need to implement this function by yourself:

https://github.com/olikraus/u8g2/wiki/Porting-to-new-MCU-platform#hardware-spi-communication
